### ISSUE TITLE
Tabs indicator being set incorrectly due to scroll bar #2

### ISSUE THREAD

**Issue by markonose** [Dogfalo/materialize#5058]
I'm referring to this post: https://github.com/Dogfalo/materialize/issues/4729

I had the same problem, it occurs because the scrollbar is not accounted for. For example the scrollbar width is 17px. When you switch from a tab that is scrollable to one that is not, the tab width is set to when the document was scrollable.

```
      // Finds right attribute for indicator based on active tab.
      // el: jQuery Object
      var calcRightPos = function(el) {
        return Math.ceil($tabs_width  - el.position().left - el.outerWidth() - $this.scrollLeft());
      };
```
If I replace $tabs_width with $this.width() it works for this example, that is, you don't get a horizontal scroll bar on the tabs, because calcRightPos doesn't return a negative value. But I don't know if it breaks anything else.

EDIT: actually it doesn't quite work as right get's styled to 1px instead of 0px, but it's an improvement over the horizontal scrollbar

**Comment by MrBLT** [comment]
I had the same issue also. Indicators were misaligned when clicking on tabs that would show enough content to force a scrollbar to be present. 

To work around this, I made a few functions that ultimately waits 200 ms after clicking the new tab, chcecking if the scrollBar state has changed, and if it has, click on the same tab again in order to re-adjust the tab indicators. 

````
var scrollBarExists = false;
function hasScrollBar() {
    return $(document).height() > $(window).height();
}
$('nav a').click(function () {
        $this = $(this);
        //wait for tab to load
        setTimeout(function () {
            //check if scrollbar value has changed
            var tmp = hasScrollBar();
            //if so, click the tab again to readjust the indicator
            if (scrollBarExists != tmp) {
                scrollBarExists = tmp;
                $this.click();
            }
        }, 200);
    });
````

**Comment by chintogtokh** [comment]
Same issue as #4400

**Comment by vanderstaaij** [comment]
This is not the same issue as #4400, as that issue is involving the scrollbar appearing because of overflow on the ul.tabs element.

This issue is different because it involves the scrollbar (e.g. on the right hand side) to pop up depending on the size of the content to show in a tab's content area. If that contains overflowing content then the scrollbar will appear in the browser. This screws up the indicator because it does not take the scrollbar pixels difference into account.

**Comment by vanderstaaij** [comment]
The issue here is that $tabs_width is set when initializing the tabs, but it is not updated when the functions for calculation of the indicator left/right positions is called.

This can be changed by replacing this:
```
var calcRightPos = function (el) {
  return Math.ceil($tabs_width - el.position().left - el[0].getBoundingClientRect().width - $this.scrollLeft());
};
```
by this:
```
var calcRightPos = function (el) {
  return Math.ceil($this.width() - el.position().left - el[0].getBoundingClientRect().width - $this.scrollLeft());
};
```

**Comment by acburst** [comment]
Fixed in 2103d82f
